gobblin-compaction/src/main/java/org/apache/gobblin/compaction/source/CompactionSource.java

```diff
@@ -128,7 +129,12 @@ public WorkUnitStream getWorkunitStream(SourceState state) {
 
       // Spawn a single thread to create work units
       new Thread(new SingleWorkUnitGeneratorService(state, prioritize(datasets, state), workUnitIterator), "SingleWorkUnitGeneratorService").start();
-      return new BasicWorkUnitStream.Builder(workUnitIterator).build();
+      WorkUnitStream workUnitStream = new BasicWorkUnitStream.Builder(workUnitIterator).build();
+      if (workUnitStream.getWorkUnits().hasNext()) {
+        // copy jars iff workunits are created
```

Comment:
small typo `if`

Reply:
is this to save time for 0 workunit jobs? add a bit more reasoning for why in comment